src/sharing/SharingScreen.js

```diff
@@ -0,0 +1,498 @@
+                narrow={narrow}
+              />
+            </AnimatedScaleComponent>
+            <Input
+              placeholder="Topic"
+              value={topic}
+              onFocus={this.focusTopic}
+              onBlur={this.blurTopic}
+              onChangeText={this.handleTopicChange}
+              editable={stream !== ''}
+            />
+            <Input
+              placeholder="Message"
+              value={message}
+              onChangeText={this.handleMessageChange}
+              multiline
+            />
+          </View>
+        </ScrollView>
+        <View style={styles.actions}>
+          <ZulipButton onPress={this.finishShare} style={styles.button} secondary text="Cancel" />
+          <ZulipButton
+            style={styles.button}
+            onPress={this.handleSend}
+            text="Send"
+            progress={sending}
+            disabled={!this.isSendButtonEnabled()}
+          />
+        </View>
+      </>
+    );
+  }
+}
+
+type ShareToPmProps = $ReadOnly<{|
+  ...$Exact<NavigationNavigatorProps<{||}, {| params: {| sharedData: SharedData |} |}>>,
+  dispatch: Dispatch,
+  auth: Auth,
+|}>;
+
+type ShareToPmState = $ReadOnly<{|
+  users: User[],
+  message: string,
+  choosingRecipients: boolean,
+  sending: boolean,
+|}>;
+
+class ShareToPmComponent extends React.Component<ShareToPmProps, ShareToPmState> {
+  constructor(props) {
+    super(props);
+    const { sharedData } = this.props.navigation.state.params;
+    this.state = {
+      users: [],
+      message: sharedData.type === 'text' ? sharedData.sharedText : '',
+      choosingRecipients: false,
+      sending: false,
+    };
+  }
+
+  setSending = () => {
+    this.setState({ sending: true });
+  };
+
+  toggleChoosingRecipients = () => {
+    this.setState(state => ({ choosingRecipients: !state.choosingRecipients }));
+  };
+
+  handleChooseRecipients = (users: Array<User>) => {
+    this.setState({ users });
+    this.toggleChoosingRecipients();
+  };
+
+  finishShare = () => {
+    const { dispatch } = this.props;
+
+    dispatch(navigateBack());
+    BackHandler.exitApp();
+  };
+
+  handleMessageChange = message => {
+    this.setState({ message });
+  };
+
+  isSendButtonEnabled = () => {
+    const { message, users } = this.state;
+    const { sharedData } = this.props.navigation.state.params;
+
+    if (sharedData.type === 'text') {
+      return message !== '' && users.length > 0;
+    }
+
+    return users.length > 0;
+  };
+
+  getUsersPreview = () => {
+    const { users } = this.state;
+
+    if (users.length === 0) {
+      return <Label text="Please choose recipients to share with" />;
+    }
+    const preview = [];
+    users.forEach((user: User) => {
+      preview.push(
+        <UserItem
+          avatarUrl={user.avatar_url}
+          email={user.email}
+          fullName={user.full_name}
+          onPress={() => {}}
+        />,
+      );
+    });
+    return preview;
+  };
+
+  handleSend = async () => {
```

Comment:
Let's put the action handlers above all the render logic -- so above `isSendButtonEnabled` as well as `getUsersPreview` / `renderUsersPreview`.

Reply:
Okay, moved `renderUsersPreview` above. Looks more uniform now :)